packages/@aws-cdk/cdk/lib/synthesis.ts

```diff
@@ -0,0 +1,98 @@
+import fs = require('fs');
+import path = require('path');
+
+export interface ISynthesisSession {
+  /**
+   * Writes a file into the synthesis session directory.
+   * @param fileName The name of the file.
+   * @param data The contents of the file.
+   */
+  writeFile(fileName: string, data: any): void;
```

Comment:
Can the standard case please be "we read and write structured data", and have maybe additional methods for writing binary files?  I don't want to have to JSON.stringify() and JSON.parse()-and-catch-and-make a descriptive error message in every call site.

Reply:
This is a "store" interface, so I added `writeJson` as well.